app/lib/devise_recognisable/guard.rb

```diff
@@ -61,19 +65,33 @@ def calculate_score_for(session)
   end
 
   # Method to check if the ip addresses are similar. Takes a session_address
-  # and returns a bool
+  # and returns a :symbol for the level of similarity.
   def compare_ip_addresses(session_address)
-    return true if session_address == @request.location.ip
-    
+    # Check if the IP is an exact match
+    return :exact_match if session_address == @request.location.ip
+
+    # Check that neither IP address is IPv6.
+    # If either is, return :ip_version_mismatch.
+    if ipv6?(@request.location.ip) || ipv6?(session_address)
+      return :ip_version_mismatch
```

Comment:
It doesn't quite feel like this is the right description. If both the incoming request _and_ session IP were IPv6 we'd say 'there's a version mismatch' but that's not true. Instead I'd just return `:complete_mismatch` as that's all we're going to do for IPv6 (e.g. either it's an exact match or nothing).

Reply:
👍  Yep